src/background/modules/AutocorrectHandler.js

```diff
@@ -26,6 +26,76 @@ let antipatterns = [];
 // Adapted from: https://github.com/mozilla/webextension-polyfill/blob/master/src/browser-polyfill.js
 const IS_CHROME = Object.getPrototypeOf(browser) !== Object.prototype;
 
+/**
+ * Traverse Trie tree of objects to create RegEx.
+ *
+ * @param {Object.<string, Object|boolean>} obj
+ * @returns {string}
+ */
+function pattern(obj) {
+    const alt = [];
+    const cc = [];
+
+    // Escape special characters
+    const regExSpecialChars = /[.*+?^${}()|[\]\\]/gu;
+
+    for (const char in obj) {
+        if (char) {
+            const achar = char.replace(regExSpecialChars, "\\$&");
+
+            if (!("" in obj[char] && Object.keys(obj[char]).length === 1)) {
```

Comment:
Is `"" in obj[char]` the `leaf note` from the function below? If so, maybe use a JS `Symbol` or other more unique and properly named indicator.

Reply:
Yes, the leaf node. I used the empty string, as it is of course not possible to have a null character in an autocorrection, but a Symbol should also be fine.

Reply:
So… can we change that? Magic strings, especially unnamed ones such as an empty string, are IMHO not clean code.

Reply:
I thought the use of the empty string here was clever and simplified the code (I guess I prefer simplicity over cleanliness), but yes, we can of course change it to a Symbol if you want...

Reply:
Well the concept of clean code says code is read like dunno 10x more than written, so I'd always advocate for cleanliness (or better: readability) over some "it's simpler to write this way" or "it is a clever hack" or so. It may be, but in 2 weeks you have forgotten :wink: 

Does not have to be a symbol, but can be something else, e.g. a helper function `isLeafNote` and then you have much more readable code:
```js
if (!isLeafNote(obj[char])) {
```

Actually, I would maybe even prefer this over a Symbol as I am writing this.

Reply:
Yes, I generally agree with you. This function of course implements a complex algorithm, so it is already somewhat confusing. However, up to a point, simplicity usually improves readability (and performance), as it reduces the amount of code one needs to read and understand.

Anyway, this line actually checks for two things, that the next node is a leaf node and that there are no other sub nodes, which means it does not need to recurse any further down the tree. Replacing it with a helper function therefore is not practical, but I added the requested JS Symbol and made another change which should hopefully improve readability.